packages/cli/src/loadSpecFromFileOrUrl.mts

```diff
@@ -33,28 +39,46 @@ export async function loadSpecFromFileOrUrl(config: IServiceConfig): Promise<IPl
 					? yaml.load(content, { json: true, schema: yaml.JSON_SCHEMA })
 					: JSON.parse(content);
 		} catch (_) {
-			throw new Error('Something went wrong while trying to parse contents');
+			throw new Error('Something went wrong while trying to parse contents from FILE');
 		}
 
 		// transform the spec using given transformer
 	} else if (config.url) {
 		// read from URL
 		logInfo('Fetching data from URL');
-		const response = await fetch(config.url);
+
+		if (!GITHUB_PAT && !process.env.CI) {
+			throw new Error(
+				'GITHUB_PAT (Personal Access Token) is not defined, please set GITHUB_PAT environment variable',
+			);
+		}
+
+		const configUrl = process.env.CI ? config.url : generateGithubApiEndpointUrl(config.url);
+		const configHeaders = process.env.CI
+			? {}
+			: { headers: { Authorization: `Bearer ${GITHUB_PAT}` } };
+
+		const response = await fetch(configUrl, {
+			...configHeaders,
+		});
+
 		const contentType = response.headers.get('Content-Type');
 		try {
 			logInfo('Parsing data from API');
+			if (contentType === 'application/json; charset=utf-8') {
```

Comment:
The `if` check earlier was for actual content of spec—whether it is in "json" format or in "yaml" format. Let's discuss this on a call.

Reply:
we need this because the else block is executed for test services which are present in example folder